primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -242,8 +241,7 @@ private static boolean isValidFileContent(PrimeApplicationContext context, FileU
             return true;
         }
 
-        String tempFilePrefix = UUID.randomUUID().toString();
-        Path tempFile = Files.createTempFile(tempFilePrefix, null);
+        Path tempFile = Files.createTempDirectory("pf-fileupload").resolve(fileName);
```

Comment:
I like the idea of having a single directory for PF (meant for internal stuff). Is there anywhere else we do this? If yes, create a global variable

Reply:
AFAIR, extension was removed on purpose to avoid content type checking based on extension. Does tika behave the same if extension present in filename?

Reply:
tika by default also tries to detect content type by file extension: https://github.com/apache/tika/blob/2.9.1/tika-java7/src/main/java/org/apache/tika/filetypedetector/TikaFileTypeDetector.java

Reply:
Indeed not great, I think there should be 2 pass then: one without extension, and another one with extension in case resolved content type is null

Reply:
I actually like your idea also a lot more than what I originally had in mind, forcing FileTypeDetector impls to do a deep inspection by default instead of going the lazy way of looking at the file extension